stl/inc/__msvc_all_public_headers.hpp

```diff
@@ -34,6 +34,7 @@
 
 #define _SILENCE_CXX17_C_HEADER_DEPRECATION_WARNING
 #define _SILENCE_CXX17_STRSTREAM_DEPRECATION_WARNING
+#define _SILENCE_EXPERIMENTAL_ERASE_DEPRECATION_WARNING

```

Comment:
The purpose of these macro definitions is to suppress errors/warnings as a consequence of including headers. (We think of these as "aggressive deprecations".) There is no header for which `_SILENCE_EXPERIMENTAL_ERASE_DEPRECATION_WARNING` suppresses such an error/warning; its definition only makes it possible to *use* the `std::experimental::erase_if` without a deprecation warning.

As such, I don't think we should make this change.

Reply:
Ahhh, thanks for the explanation.

Reply:
Agreed. By not silencing the `experimental::erase_if` deprecation here, our test coverage verifies that nothing in the STL is using `experimental::erase_if`. (The tests specifically for `experimental::erase_if` then silence the warning.) Of course, there's no way you could have known that without seeing our tests (which will be available soon!). :-)